projects/core/src/lazy-loading/lazy-modules.service.ts

```diff
@@ -114,6 +126,65 @@ export class LazyModulesService implements OnDestroy {
     );
   }
 
+  /**
+   * The purpose of this function is to run MODULE_INITIALIZER logic that can be provided
+   * by a lazy loaded module.  The module is recieved as a function parameter.
+   * This function returns an Observable to the module reference passed as an argument.
+   *
+   * @param {NgModuleRef<any>} moduleRef
+   *
+   * @returns {Observable<NgModuleRef<any>>}
+   */
+
+  public runModuleInitializersForModule(
+    moduleRef: NgModuleRef<any>
+  ): Observable<NgModuleRef<any>> {
+    const moduleInits: any[] = moduleRef.injector.get<any[]>(
+      MODULE_INITIALIZER,
+      [],
+      InjectFlags.Self
+    );
+    const asyncInitPromises = this.runModuleInitializerFunctions(moduleInits);
+    if (asyncInitPromises.length) {
+      return from(Promise.all(asyncInitPromises)).pipe(
+        switchMapTo(of(moduleRef))
+      );
+    } else {
+      return of(moduleRef);
+    }
+  }
+  /**
+   * This function accepts an array of functions and runs them all. For each function that returns a promise,
+   * the resulting promise is stored in an array of promises.  That array of promises is returned.
+   * It is not required for the functions to return a Promise.  All functions are run.  The return values
+   * that are not a Promise are simply not stored and returned.
+   *
+   * @param {(() => any)[]} initFunctions An array of functions too be run.
+   *
+   * @return {Promise<any>[]} An array of Promise returned by the functions, if any,
+   */
+  public runModuleInitializerFunctions(
+    initFunctions: (() => any)[]
+  ): Promise<any>[] {
+    const initPromises: Promise<any>[] = [];
+    if (initFunctions) {
+      for (let i = 0; i < initFunctions.length; i++) {
+        const initResult = initFunctions[i]();
+        if (this.isObjectPromise(initResult)) {
+          initPromises.push(initResult);
+        }
+      }
+    }
+    return initPromises;
+  }
+
+  /**
+   * Determine if the argument is shaped like a Promise
+   */
+  private isObjectPromise<T = any>(obj: any): obj is Promise<T> {
+    return !!obj && typeof obj.then === 'function';
+  }
```

Comment:
This util function did not seem to be public from angular.

Reply:
👍